### PR TITLE
Support h.264 video decoding on web

### PR DESCRIPTION
**Pull request by jprochazk**
### What

* Closes https://github.com/rerun-io/rerun/issues/7375

- The seeking logic now searches for the requested segment by the presentation/composition timestamp passed in by the user. This way if a user requests frame at 12s they will get the frame which will be _presented_ at 12s, not _decoded_ at 12s. See [I/P/B frame terminology](https://en.wikipedia.org/wiki/Video_compression_picture_types) in video encoding for more information as to why this distinction matters.
- `EncodedVideoChunk`s are now given the _composition timestamp_ of the sample instead of its _decode timestamp_, so the output `VideoFrame` objects are in _composition timestamp_ order.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7511?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7511?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7511)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_renderer/src/video/decoder/web.rs [hunk 1]
```diff
@@ -200,47 +200,80 @@ impl VideoDecoder {
...
+
+    fn enqueue_requested_segments(
+        &mut self,
+        presentation_timestamp: Time,
+    ) -> Result<(), DecodingError> {
+        // Some terminology:
+        //   - presentation timestamp = composition timestamp
+        //     = the time at which the frame should be shown
+        //   - decode timestamp
+        //     = determines the decoding order of samples
```
Comment (Wumpf) [inline crates/viewer/re_renderer/src/video/decoder/web.rs:224]: good call to document that here, makes this stuff a lot more maintainable 👍

File: crates/viewer/re_renderer/src/video/decoder/web.rs [hunk 2]
```diff
@@ -200,47 +200,80 @@ impl VideoDecoder {
...
+            &self.data.samples,
+            |sample| sample.decode_timestamp,
+            &presentation_timestamp,
+        ) else {
+            return Err(DecodingError::EmptyVideo);
+        };
+
+        // 2. Search _backwards_, starting at `decode_sample_idx`, looking for
+        //    the first sample where `sample.composition_timestamp <= presentation_timestamp`.
+        //    This is the sample the user requested.
```
Comment (Wumpf) [inline crates/viewer/re_renderer/src/video/decoder/web.rs:243]: nit: "uhm, actually" the user technically doesn't request a sample. A single sample is not able to represent the user's request

... but I don't know how to describe it better either right now. Fine to leave it this way I think

**Comment by jprochazk** [comment]
Seems like the lint failure comes from main

**Review by Wumpf — APPROVED** [review]
really well document, also like it a lot how you split up the concerns of frame_at further and made everything more precise!

File: crates/viewer/re_renderer/src/video/decoder/web.rs (hunk 2 above)
Comment (jprochazk) [reply crates/viewer/re_renderer/src/video/decoder/web.rs]: Changed it to "This is the sample which when decoded will be presented at the timestamp the user requested."

**Pull request merged**